# Scripts-pipeline-SVs  

A bioinformatics protocol designed for processing VCF files generated by three structural variant (SV) callers in whole-genome sequencing (WGS) samples. The pipeline consolidates the detected variants into a single BED file for downstream analysis.  

---

## How to run this pipeline?  

The files from the sample you wish to analyze must be stored in the same directory, organized as follows:  

**Example directory structure:**  

- /0032_0032/
    - Manta/
          Manta_0032.vcf
    - delly/
          delly_0032.bcf
    - GRIDSS/
          GRIDSS_0032.vcf

Once organized, simply execute the following command:  

```bash
./pipe_SV.sh path/to/directory/0032_0032
> [!IMPORTANT]
> Do not include the trailing / when specifying the directory path.
## Esquema funcionamiento general ##

![. Flujo de trabajo de los scripts incorporados para procesar los VCF de cada programa y obtener un único archivo con
todas las variantes incluyendo las detectadas por varios programas. Los scripts destinados al procesado de los VCF se recuadran en
rojo y los destinados a la fusión de las variantes en un solo archivo en naranja. DEL = Deleción, INS= Inserción, DUP =
Duplicación, TRA = Translocación, INV= Inversión.
](https://github.com/paudbr2002/Scripts-pipeline-SVs/blob/main/Esquema_pipeline.png)

Step-by-step overview:

**1.Filtering and preparation**

- Directories are created to filter variants with “PASS” quality.

- VCF files are converted into BED format.

**2.Processing per SV type**

- Scripts process formatted files for each caller and SV type.

- SVs are separated by genotype, then sorted and grouped by coordinates.

- Specific tools handle certain SVs (e.g., bedtools merge for inversions and deletions).

**3.Annotation and consolidation**

- Necessary annotations are added.

- All results are combined into a single final file with a specific format (see Supplementary Table 1).

- The final file contains all variants grouped by coordinates, with information on overlaps across callers to create a consensus view.
Software versions

bcftools 1.15

bedtools 2.27.1

BEDOPS v2.4.41

Python 3.12.3

Bash 5.0.17

R 4.4.0

## Author ##
Paula de Blas Rioja 2024 (pau.dbr2002@gmail.com)
